File: backend/app/model/dto/response/activityDistributionDto.py
from pydantic import BaseModel, Field


class ActivityDistributionItemDto(BaseModel):
    """Single activity distribution item"""
    activity_label: str = Field(..., description="Activity type label")
    total_seconds: float = Field(..., description="Total seconds spent on this activity")
    total_minutes: float = Field(..., description="Total minutes spent on this activity")
    total_hours: float = Field(..., description="Total hours spent on this activity")
    percentage: float = Field(..., description="Percentage of time spent on this activity")


class ActivityDistributionResponseDto(BaseModel):
    """Response containing activity distribution for a time range"""
    user_id: str = Field(..., description="User ID")
    start_time: str = Field(..., description="Start of time range (ISO format)")
    end_time: str = Field(..., description="End of time range (ISO format)")
    activities: list[ActivityDistributionItemDto] = Field(..., description="List of activities with their distribution")
    total_hours: float = Field(..., description="Total hours of tracked activity")

File: mobile/src/main/java/com/example/harbit/data/remote/dto/ActivityDistributionDto.kt
package com.example.harbit.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ActivityDistributionItemDto(
    @SerialName("activity_label")
    val activityLabel: String,
    
    @SerialName("total_seconds")
    val totalSeconds: Double,
    
    @SerialName("total_minutes")
    val totalMinutes: Double,
    
    @SerialName("total_hours")
    val totalHours: Double,
    
    @SerialName("percentage")
    val percentage: Double
)

@Serializable
data class ActivityDistributionResponseDto(
    @SerialName("user_id")
    val userId: String,
    
    @SerialName("start_time")
    val startTime: String,
    
    @SerialName("end_time")
    val endTime: String,
    
    @SerialName("activities")
    val activities: List<ActivityDistributionItemDto>,
    
    @SerialName("total_hours")
    val totalHours: Double
)
